### ISSUE TITLE
ABuildHasher isn't returning consistent results

### ISSUE THREAD

**Issue by Amanieu** [tkaitchuck/aHash#13]
A `BuildHasher` needs to create `Hashers` with the same seed so that they return consistent results. The seed should be randomized when the `BuildHasher` is created, not when the `Hasher` is built.

**Comment by tkaitchuck** [comment]
That's definitely a bug. I suspect it's being caused when the HashBuilder is being moved between invocations. To confirm, was this the case in your test?

**Comment by Amanieu** [comment]
Change this line in my `ahash` branch on Amanieu/hashbrown from:
```rust
pub type DefaultHashBuilder = core::hash::BuildHasherDefault<ahash::AHasher>;
```
to:
```rust
pub type DefaultHashBuilder = ahash::ABuildHasher;
```
Then run the tests and you will see many failures.

**Comment by tkaitchuck** [comment]
This should be fixed in 0.2.4.
I have run the tests against Amanieu/hashbrown
